.github/workflows/publish.yml

```diff
@@ -0,0 +1,87 @@
+name: publish artifacts
+on:
+  push:
+    branches:
+      - 'main'
+  workflow_dispatch:
+jobs:
+  build-napi-linux:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: napi build
+        run: yarn install && yarn build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: linux-napi
+          path: ./*.node
+          if-no-files-found: error
+  build-napi-macos:
+    runs-on: macos-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: napi build
+        run: yarn install && yarn build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: macos-napi
+          path: ./*.node
+          if-no-files-found: error
+  build-napi-win:
+    runs-on: windows-latest
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v3
+      - name: napi build
+        run: yarn install && yarn build
+      - uses: actions/upload-artifact@v3
+        with:
+          name: windows-napi
+          path: ./*.node
+          if-no-files-found: error
+      - uses: actions/upload-artifact@v3
+        with:
+          name: js-files
+          path: |
+            index.d.ts
+            index.js
+          if-no-files-found: error
+  publish:
```

Comment:
This job should be conditional based on if the version was updated in source. Otherwise this job will try to publish every commit and fail when there is a conflicting version already in npm

If you want reference on how to do that, I set this up for ori here:
https://github.com/microsoft/ori/blob/main/.github/workflows/publish-prod.yml#L19-L35

script is here
- https://github.com/microsoft/ori/blob/main/scripts/is-any-new-version.sh
- https://github.com/microsoft/ori/blob/main/scripts/_isnewversion.sh